src/fn/create-users.js

```diff
@@ -1,52 +1,92 @@
-            info('Creating user', username);
-            return request({
-              uri: `${instanceUrl}/api/v1/users`,
-              method: 'POST',
-              json: true,
-              body: requestObject,
-            });
-          });
-      }, Promise.resolve());
+const nestPrefixedProperties = (obj, name) => {
+  const nested = {};
+  const prefix = `${name}.`;
+  Object
+    .keys(obj)
+    .filter(key => key.startsWith(prefix))
+    .forEach(key => {
+      nested[key.substring(prefix.length)] = obj[key];
+      delete obj[key];
     });
+  return obj[name] || nested;
 };
 
-function prefixedProperties (cols, row, prefix) {
-  const indices = {};
-  cols.forEach(col => {
-    if (col.startsWith(prefix)) {
-      indices[col.substring(prefix.length)] = row[cols.indexOf(col)];
-    }
+const getUsersData = (csvData) => {
+  const users = csvParse(csvData, { columns: true });
+  users.forEach(user => {
+    user.contact = nestPrefixedProperties(user, 'contact');
+    user.place = nestPrefixedProperties(user, 'place');
+    user.roles = user.roles && user.roles.split(':');
   });
-  return indices;
-}
+  return users;
+};
+
+const getUserInfo = async (instanceUrl, user) => {
+  const getId = (obj) => typeof obj === 'string' ? obj : obj._id;
+  const facilityId = getId(user.place);
+  if (!facilityId) {
+    // new place - nothing to check
+    return;
+  }
+
+  const params = {
+    facility_id: facilityId,
+    role: JSON.stringify(user.roles),
+    contact: getId(user.contact),
+  };
+
+  info(`Requesting user-info for "${user.username}"`);
+  let result;
+  try {
+    result = await request.get(`${instanceUrl}/api/v1/users-info`, { qs: params, json: true });
+  } catch (err) {
+    // we can safely ignore some errors
+    // - 404: This endpoint was only added in 3.7
+    // - 400: The endpoint throws an error if the requested roles are "online"
+    // - 400: Missing facility or role, the corresponding user create request will fail
+    if (err.statusCode !== 404 && err.statusCode !== 400) {
+      throw err;
+    }
+  }
+  return result;
+};
+
+const createUser = async (instanceUrl, user) => {
+  return request.post(`${instanceUrl}/api/v1/users`, { json: true, body: user });
+};
+
+module.exports = async (projectDir, couchUrl) => {
+  if(!couchUrl) {
+    throw new Error('Server URL must be defined to use this function.');
+  }
+
+  const instanceUrl = couchUrl.replace(/\/medic$/, '');
+  const csvPath = `${projectDir}/users.csv`;
+  if(!fs.exists(csvPath)) {
+    throw new Error(`User csv file not found at ${csvPath}`);
+  }
+
+  const users = getUsersData(fs.read(csvPath));
+  for (let user of users) {
+    const userInfo = await getUserInfo(instanceUrl, user);
+    if (userInfo && userInfo.warn) {
+      warn(`The user "${user.username}" would replicate ${userInfo.total_docs}, which is above the recommended limit of ${userInfo.limit}. Are you sure you want to continue?`);
+      if(!readline.keyInYN()) {
```

Comment:
Should we prompt once per user, or could we log all the warnings and then prompt once? It seems like this current pattern could be overbearing if you actually wanted to ignore the warnings while uploading lots of users.

Reply:
I think the idea is to _not_ ignore the warnings. Misconfigured prove to be really problematic in projects, which is why this feature is being implemented. 
But sure, I guess we can log once for all users. :)